src/buffer/buffer.rs

```diff
@@ -370,6 +449,52 @@ impl Buffer {
     }
 }
 
+impl<P: Into<Position>> Index<P> for Buffer {
```

Comment:
Increase readability by clarifying what exactly is implemented here inline. When first reading this I thought there is yet another ratatui trait. (same for `IndexMut`)

```suggestion
impl<P: Into<Position>> std::ops::Index<P> for Buffer {
```

Reply:
I changed this back to just Index as the trait has to be in scope for links to a trait defined method in a doc comment to resolve correctly.